### ISSUE TITLE
Cannot load parameters from empty string path

### ISSUE THREAD

**Issue by MrCochese** [aws/aws-dotnet-extensions-configuration#40]
Expected: Setting the Path option to empty string loads all parameters.
Actual: Path cannot be null exception.

Setting path to the minimum "/" loads all parameters, but trims the first character from each of the keys.

**Comment by klaytaybai** [comment]
Hi @MrCochese, could you pleas explain why you think an empty string should load all parameters?  This seems to be by design.  As you indicate, "/" is a valid base path.

I wasn't able to reproduce the issue you are seeing with the keys of the parameters losing the first character.  Could you please provide reproduction steps, including which version you are using?

**Comment by MrCochese** [comment]
The keys in my store do not begin with a prefix or base path of any kind,
therefore this library is of no use if it cannot accept an empty path. I'm
not even sure why this would be the case, as the AWS SDK is perfectly
capable of loading all keys.

**Comment by KenHundley** [comment]
Hey @MrCochese, the project uses the GetParametersByPath call and translates the hierarchy into configuration data. Per the SSM Param Store documentation hierarchies are required to start with a "/"

```
get-parameters-by-path
 "--path" (string)

   The hierarchy for the parameter. Hierarchies start with a forward
   slash (/) and end with the parameter name. A parameter name
   hierarchy can have a maximum of 15 levels. Here is an example of a
   hierarchy: "/Finance/Prod/IAD/WinServ2016/license33"
```

I don't see any reason why it shouldn't handle "/" but I don't believe it was tested that way as it would potentially returns hundreds, thousands, or more parameter results. 

I'll try to take a look a little later today to see why it could be stripping off the first character when you only provide "/". It should be a simple tweak to resolve this.

**Comment by MrCochese** [comment]
Thanks for looking. Just checked this again, as I read the Simple Systems
Manager page on the pathing and thought maybe I'd made a mistake, as in the
web interface it actually is possible to path to all parameters with just
"/" - but using the library (1.1.1 with AWSDK at 3.3.1010.9) it is
definitely loading in every parameter with the initial character stripped.

I imagine at some point my organisation may want to consider using pathing
to organise our keys, but at the moment it's not really on the cards and
our set is still quite small.

**Comment by KenHundley** [comment]
I found the source of the issue. 

The code expected all returned params to match the Path provided and strips off that path before processing. However, `/` is an exception where it returns everything, even if the param doesn't actually start with `/`. So when it went to strip the `/` from the beginning of the returned params it ended up stripping off the first char for those that didn't start with `/`.

I'll make a PR that prevents this from occuring for `/`

**Comment by normj** [comment]
@KenHundley PR was merged and released as part of version 1.2.0.
